Userland/Services/SQLServer/DatabaseConnection.cpp

```diff
@@ -41,7 +41,7 @@ DatabaseConnection::DatabaseConnection(String database_name, int client_id)
     deferred_invoke([&](Object&) {
         m_database = SQL::Database::construct(String::formatted("/home/anon/sql/{}.db", m_database_name));
         m_accept_statements = true;
-        auto client_connection = ClientConnection::client_connection_for(client_id);
+        auto client_connection = ClientConnection::client_connection_for(m_client_id);
```

Comment:
I'm with @rstefanic and don't understand either what is going on. As far as I understand, ``client_id`` is captured in the lambda, because everything is captured, so I don't understand why it changes in between the lambda being created and when it is executed.

Unless of course it's captured by reference (which it is) and the value of the object pointed by the reference changes. 

🤔

Let me do some testing

Reply:
Yeah if I explicitly capture ``this`` and ``client_id`` it works as well. But let's go with your patch. Saves busy work...

Reply:
Yeah that capture-by-reference is not safe, since it's capturing a value on the stack. By the time the `deferred_invoke` callback fires (via the event loop), `client_id` is no longer on the stack.

To avoid issues like this, I recommend not using `[&]` captures for asynchronously invoked callbacks. Using `[this]` and `m_client_id` seems like the best approach here. :)

Reply:
@awesomekling Good to know. I'll change them all in my next round of patches.